examples/gno.land/r/gov/dao/dao.gno

```diff
@@ -8,38 +8,58 @@ import (
 	pproposal "gno.land/p/gov/proposal"
 )
 
-var proposals = make([]*proposal, 0)
+var (
+	proposals = make([]*proposal, 0)
+	members   = make([]std.Address, 0) // XXX: these should be pointers to avoid data duplication. Not possible due to VM bugs
+)
+
+const (
+	msgMissingExecutor = "missing proposal executor"
+	msgPropExecuted    = "prop already executed"
+	msgPropExpired     = "prop is expired"
+	msgPropInactive    = "prop is not active anymore"
+	msgPropActive      = "prop is still active"
+	msgPropNotAccepted = "prop is not accepted"
 
-// XXX var members ...
+	msgCallerNotAMember = "caller is not member of govdao"
+	msgProposalNotFound = "proposal not found"
+)
 
 type proposal struct {
-	idx      int
```

Comment:
What about keeping it?

I agree it's not needed for the lookup, but I think it makes sense to keep it so that a proposal object is marshallable using Guilhem's upcoming native JSON support (#1776). Additionally, a pointer to a proposal could be kept by other contracts, and we may imagine having things like `prop.Render()`.

Reply:
I understand your concerns, but having duplicated data from my experience leads to shady errors :D.

If we want to serialize a proposal as JSON, we can do the following when needed (as an example):

```go

type proposalJSON struct {
 	proposal
	idx int
}

```